source/vir/src/prune.rs

```diff
@@ -266,7 +266,10 @@ pub fn prune_krate_for_module(krate: &Krate, module: &Path) -> (Krate, Vec<MonoT
     for f in &functions {
         function_map.insert(f.x.name.clone(), f.clone());
         if let FunctionKind::TraitMethodImpl { method, datatype, .. } = &f.x.kind {
-            method_map.insert((datatype.clone(), method.clone()), f.x.name.clone());
+            // TODO(tchajed): construct a path for a method impl
```

Comment:
I don't think you'll need this `method_map` to have non-`Datatype` types.  If it's not a `Datatype`, then `reached_methods` can just assume that the type is reached, so all the type's methods should be reached.  In other words, for `Datatype`s `D` we have:
```
    // If:
    // - we reach both D and T.f
    // - and D implements T.f with D.f
    // add D.f
```
and for non-`Datatype`s `D`, we have:
```
    // If:
    // - we reach T.f
    // - and D implements T.f with D.f
    // add D.f
```